app/graphql/types/extensions/paginate.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Types
+  module Extensions
+    class Paginate < GraphQL::Schema::FieldExtension
+      def apply
+        field.argument :limit, "Types::Scalar::Limit", required: true
+        field.argument :offset, "Types::Scalar::Offset", required: false
+      end
+
+      def resolve(object:, arguments:, context:)
+        next_args = arguments.dup
+        next_args.delete(:limit)
+        next_args.delete(:offset)
+        yield(object, arguments)
+      end
+
+      def after_resolve(object:, arguments:, value:, context:, memo:)
+        raise GraphQL::ExecutionError, 'Paginate extension can only be used on ActiveRecord:Relation objects' unless value.is_a?(ActiveRecord::Relation)
```

Comment:
If we want to use this later with Digital Object search result sets, they won't be ActiveRecord::Relation objects.  We can cross that bridge when we get to it though.  One alternative might be to just make sure that the `value` responds to the expected methods.  Or maybe we'll end up handling pagination separately for that.

Reply:
Ah, I see Types::Extensions:: SolrSearch below, and how it's easy enough to introduce another FieldExtension later on, so I'm not worried about Digital Object results -- which could be Rsolr search result sets, but may actually be an aggregation of actual DigitalObject::Base subclass objects instantiated from the solr docs.